clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -207,14 +207,22 @@ public UpdateMetadataRequest build(short version) {
 
         @Override
         public String toString() {
-            StringBuilder bld = new StringBuilder();
-            bld.append("(type: UpdateMetadataRequest=").
-                append(", controllerId=").append(controllerId).
-                append(", controllerEpoch=").append(controllerEpoch).
-                append(", brokerEpoch=").append(brokerEpoch).
-                append(", partitionStates=").append(partitionStates).
-                append(", liveBrokers=").append(Utils.join(liveBrokers, ", ")).
-                append(")");
+          StringBuilder bld = new StringBuilder();
+          // HOTFIX: LIKAFKA-24478
+          // large cluster with large metadata can create really large string
+          // potentially causing OOM
+          bld.append("(type: UpdateMetadataRequest=").
+              append(", controllerId=").append(controllerId).
+              append(", controllerEpoch=").append(controllerEpoch).
+              append(", brokerEpoch=").append(brokerEpoch).
+              append(")");
+              // bld.append("(type: UpdateMetadataRequest=").
+              //   append(", controllerId=").append(controllerId).
+              //   append(", controllerEpoch=").append(controllerEpoch).
+              //   append(", brokerEpoch=").append(brokerEpoch).
+              //   append(", partitionStates=").append(partitionStates).
+              //   append(", liveBrokers=").append(Utils.join(liveBrokers, ", ")).
```

Comment:
I think the "liveBrokers" list is useful while troubleshooting. Is the list of broker large or is it the partition states?

Reply:
`partitionsStates` is the largest in relation to anything else so definitely we can keep the `liveBrokers` list.
Wondering how does the alive broker list help here?

Reply:
It indicates the list of brokers that the controller thinks is alive and hence, it is sending it out `UpdateMetadataRequest` to those brokers. There is no other way to find out which and all brokers should have received this update. 

I think the latest version in apache improves this kind of logging by grouping the partitions per topic or something. Need to check that. But for this particular request, the partition state feels like an overkill.